libs/sql-schema-describer/src/mssql.rs

```diff
@@ -340,6 +382,28 @@ impl SqlSchemaDescriber {
         for row in rows {
             debug!("Got index row: {:#?}", row);
 
+            // Skip the index if it contains a partial index condition other
+            // than `[col] IS NOT NULL`.
+            //
+            // SQL Server allows only one null item per unique column by
+            // default, so we'll be having a partial index condition to filter
+            // nulls out from the index.
+            if let Some(filter_definition) = row
+                .get("filter_definition")
+                .and_then(|v| v.as_str())
+                .and_then(|v| DEFAULT_DB_GEN.captures_iter(v).next())
+                .and_then(|v| v.get(0))
+                .map(|v| v.as_str().to_string())
+            {
+                let is_user_defined_partial_index = filter_definition
+                    .split(" AND ")
+                    .any(|filter| NULL_PARTIAL_INDEX.captures_iter(filter).next().is_none());
+
+                if is_user_defined_partial_index {
+                    continue;
+                }
+            };
```

Comment:
should we note that we are planning to revert this, here?

Reply:
https://github.com/prisma/prisma-engines/issues/1220